app/stac_api/pgtriggers.py

```diff
@@ -0,0 +1,290 @@
+import pgtrigger
+
+AUTO_VARIABLES_FUNC = """
+-- update auto variables
+NEW.etag = gen_random_uuid();
+NEW.updated = now();
+
+RAISE INFO 'Updated auto fields of %.id=% due to table updates.', TG_TABLE_NAME, NEW.id;
+
+RETURN NEW;
+"""
+
+
+def generate_child_triggers(parent_name, child_name):
+    '''Function that generates two triggers for parent child relation.
+
+    The triggers update the `updated` and `etag` fields of the parent when a child gets inserted,
+    updated or deleted.
+
+    Returns: tuple
+        Tuple of Trigger
+    '''
+    child_update_func = """
+    -- update related {parent_name}
+    UPDATE stac_api_{parent_name} SET
+        updated = now(),
+        etag = public.gen_random_uuid()
+    WHERE id = {child_obj}.{parent_name}_id;
+
+    RAISE INFO 'Parent table {parent_name}.id=% auto fields updated due to child {child_name}.id=% updates.',
+        {child_obj}.{parent_name}_id, {child_obj}.id;
+
+    RETURN {child_obj};
+    """
+    return (
+        pgtrigger.Trigger(
+            name=f"add_{parent_name}_child_trigger",
+            operation=pgtrigger.Insert,
+            when=pgtrigger.After,
+            func=child_update_func.format(
+                parent_name=parent_name, child_obj="NEW", child_name=child_name
+            )
+        ),
+        pgtrigger.Trigger(
+            name=f"update_{parent_name}_child_trigger",
+            operation=pgtrigger.Update,
+            when=pgtrigger.After,
+            condition=pgtrigger.Condition('OLD.* IS DISTINCT FROM NEW.*'),
+            func=child_update_func.format(
+                parent_name=parent_name, child_obj="NEW", child_name=child_name
+            )
+        ),
+        pgtrigger.Trigger(
+            name=f"del_{parent_name}_child_trigger",
+            operation=pgtrigger.Delete,
+            when=pgtrigger.After,
+            func=child_update_func.format(
+                parent_name=parent_name, child_obj="OLD", child_name=child_name
+            )
+        )
+    )
+
+
+def generates_asset_triggers():
+    '''Generates Asset triggers
+
+    Those triggers update the `updated` and `etag` fields of the assets and their parents on
+    update, insert or delete. It also update the collection summaries.
+
+    Returns: tuple
+        tuple for all needed triggers
+    '''
+
+    class UpdateCollectionSummariesTrigger(pgtrigger.Trigger):
+        when = pgtrigger.After
+        func = '''
+        IF (TG_OP != 'DELETE') THEN
```

Comment:
```SQL
IF (TG_OP != 'DELETE') THEN
    asset_instance = NEW;
ELSE
    asset_instance = OLD;
END IF;
```

can be simplified to
```SQL
asset_instance = COALESCE(NEW, OLD);
```

found here: https://www.cybertec-postgresql.com/en/what-to-return-from-a-postgresql-row-level-trigger/